src/dmd/dsymbolsem.d

```diff
@@ -4851,6 +4863,13 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
         cldec.dtor = buildDtor(cldec, sc2);
 
+        if (cldec.classKind == ClassKind.cpp && cldec.cppDtorVtblIndex != -1)
+        {
+            // now we've built the aggregate destructor, we'll make it virtual and assign it to the reserved vtable slot
+            cldec.dtor.vtblIndex = cldec.cppDtorVtblIndex;
```

Comment:
Why is `cppDtorVtblIndex` needed, why can't `cldec.dtor.vtblIndex` be used directly?

Reply:
I mean, you can see on this line that `cldec.dtor.vtblIndex` is being assigned :)
That function was only created on the line above, but its virtual slot was reserved way back when processing members...